uat/custom-components/client-java-sdk/src/main/java/com/aws/greengrass/testing/mqtt5/client/sdkmqtt/MqttConnectionImpl.java

```diff
@@ -36,26 +38,30 @@
 
 import java.util.List;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
 import java.util.concurrent.TimeUnit;
 import java.util.concurrent.atomic.AtomicBoolean;
-import java.util.function.BiConsumer;
 import java.util.stream.Collectors;
 
 /**
  * Implementation of MQTT5 connection.
  */
 public class MqttConnectionImpl implements MqttConnection {
+    private static final long MIN_SHUTDOWN_NS = 200_000_000;      // 200ms
+
     private static final Logger logger = LogManager.getLogger(MqttConnectionImpl.class);
 
     private final AtomicBoolean isClosing = new AtomicBoolean();
     private final AtomicBoolean isConnected = new AtomicBoolean();
```

Comment:
Not sure if we need atomic variables here unless we don't modify them concurrently in different threads

Reply:
Callbacks are called from MQTT client core, I have no information which thread call they. I just use AtomicBoolean to check connection states due to Boolean is immutable.